src/app/providers/NFT/BoosterContext.tsx

```diff
@@ -0,0 +1,92 @@
+'use client'
+import { AVERAGE_BLOCKTIME, NFT_BOOSTER_DATA_URL } from '@/lib/constants'
+import { axiosInstance } from '@/lib/utils'
+import { useQuery } from '@tanstack/react-query'
+import { createContext, ReactNode, useContext, useMemo } from 'react'
+import { Address } from 'viem'
+import { useAccount } from 'wagmi'
+
+interface HolderRewards {
+  estimatedRBTCRewards: bigint
+  estimatedRIFRewards: bigint
+  boostedRBTCRewards: bigint
+  boostedRIFRewards: bigint
+}
+
+interface Holders {
+  [holderAddress: string]: HolderRewards
+}
+
+interface BoostData {
+  nftContractAddress: Address
+  boostPercentage: bigint
+  calculationBlock: bigint
+  holders: Holders
+}
+interface NFTBoosterContext {
+  boostData?: BoostData
+  isLoading?: boolean
+  error?: Error | null
+  currentBoost?: HolderRewards
+}
+
+const BoosterContext = createContext<NFTBoosterContext>({
+  boostData: {} as BoostData,
+})
+
+interface BoosterContextProviderProps {
+  children: ReactNode
+}
+export const BoosterProvider = ({ children }: BoosterContextProviderProps) => {
+  const { address } = useAccount()
+  const { data: boostData, isLoading, error } = useFetchBoostData()
+  const value = useMemo(() => {
+    if (!boostData && !isLoading && !error) {
+      return {}
+    }
+
+    if (!address) {
+      return { boostData, isLoading, error }
+    }
+
+    const currentBoost = boostData?.holders[address] || {
```

Comment:
we may need to differentiate the situation where the user doesn't have the NFT and the user has the NFT but doesn't have allocation.

Reply:
done